packages/ui/src/components/Stepper/Stepper.tsx

```diff
@@ -0,0 +1,89 @@
+import React from "react";
+import { useIntl } from "react-intl";
+
+import { uiMessages } from "@gc-digital-talent/i18n";
+
+import Heading, { HeadingLevel } from "../Heading";
+
+import Step from "./Step";
+import { StepState, StepType } from "./types";
+
+const deriveStepState = (
+  stepIndex: number,
+  currentIndex: number,
+  completed = false,
+  error = false,
+): StepState => {
+  if (stepIndex > currentIndex) {
+    return "disabled";
+  }
+
+  if (stepIndex === currentIndex) {
+    return "active";
+  }
+
+  if (completed && !error) {
+    return "completed";
+  }
+
+  return "error";
+};
```

Comment:
I don't think this is quite right. It's possible to complete a step, return to a previous step, and then go back to the later step, so there's no reason steps greater than current must be disabled.

Reply:
Thinking through it a bit more, I think this is the logic:
- If `error ==  true`, it shows red and is not disabled.
- If `complete == true && error !== true`, it shows green and is not disabled.
- The first non-complete step is the active step. It shows red if `error == true`, otherwise purple.
- Other non-complete, non-error steps are disabled.
- The current step is unrelated to the active step. The current step is bolded.
- If all steps are complete, no active step(?)

When we initially place the Stepper component on the page, currentIndex will also likely be the active step, but that may change as you navigate around.

Reply:
Hmm, one thing here - the purple indicator is telling the user where they are not, the most recently completed step. If the user starts a step and navigates away to a previous step, the step they've landed on should be purple, and the last step with information entered should be green (if it was completed) or red (if there's a validation error, which there likely will be if the form is half filled).

Reply:
Okay, I think I _may_ have gotten it right in [update stepper state logic](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5902/commits/dfd0727cac007fc6417415e8c8d7b46ca8051459) based on this feedback.